andela/apollo-ah-backend
# #164198185 delivers creating articles feature

#### What does this PR do?

- provide a route for creating articles

#### Description of Task to be completed?

- [x] create article models
- [x] create a route for posting articles: ```/api/v1/articles/```
- [x] ensure the article is created 
- [x] provide a response if the article was created or not
- [x] ensure all test passes
- [x] prevent a user from creating articles with similar titles

#### How should this be manually tested?

- clone/download this repository
- install node on your system if you haven't
- change directory into the now downloaded/clone folder
- create a ```.env``` file from the available ```.env.examples``` file
- create a ```.env.test``` file with the same procedure as above
- change the ```DB_DATABASE``` value in the .env.test like so:
```
DB_DATABASE=anythingyouwishtocallit

```
- run npm install
- run npm run migration
-  create a user from the user route ```/api/v1/users/```
- visit the route and post the following data in json format:

```
title: title of your article
body: body of the article
description: a short description of the article
```
- confirm the response from the server

#### Any background context you want to provide?

- The above procedure assumes a little familiarity with programming.
- The above procedure absolutely requires creation of a user

#### What are the relevant pivotal tracker stories?

[#164198185](https://www.pivotaltracker.com/story/show/164198185)
